### PR TITLE
Exclude hidden fields from index and view template generation

### PR DESCRIPTION
**Pull request by othercorey**
Closes https://github.com/cakephp/bake/issues/717

Defaults to the same hidden fields as ModelCommand if the entity class is empty.

### CODE REVIEW

File: src/Command/TemplateCommand.php [hunk 1]
```diff
@@ -278,6 +285,7 @@ protected function _loadController(ConsoleIo $io): array
             $singularHumanName = $this->_singularHumanName($this->controllerName);
             $schema = $modelObject->getSchema();
             $fields = $schema->columns();
+            $hidden = $modelObject->newEmptyEntity()->getHidden() ?: ['token', 'password', 'passwd'];
```
Comment (markstory) [inline src/Command/TemplateCommand.php:288]: Should we have a constant for the list of fields hidden by default? That would prevent divergence in the future.

**Review by markstory — APPROVED** [review]
Nice work!

File: src/Command/TemplateCommand.php (hunk 1 above)
Comment (othercorey) [reply src/Command/TemplateCommand.php:288]: I was just going to rely on the Entity class, but we don't have them in the test suite. Should we exclude anything if the user doesn't have an entity class defined yet?

Comment (markstory) [reply src/Command/TemplateCommand.php:288]: > Should we exclude anything if the user doesn't have an entity class defined yet?

You currently have the entity taking ownership of the hidden fields list which is :+1:. We can, and should provide reasonable fallbacks if the developer hasn't defined any fields as hidden. The scenario where a developer willingly wants to show `password` will be exceptionally rare, and defaulting to keeping the developer from making a big mistakes is a defensible position for us to take.

**Pull request merged**